group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2621,26 +3497,140 @@ public void replay(
     public void replay(
         ConsumerGroupCurrentMemberAssignmentKey key,
         ConsumerGroupCurrentMemberAssignmentValue value
+    ) {
+        replay(key, value, CONSUMER);
+    }
+
+    /**
+     * Replays ConsumerGroupCurrentMemberAssignmentKey/Value to update the hard state of
+     * the consumer group. It updates the assignment of a member or deletes it.
+     *
+     * @param key   A ConsumerGroupCurrentMemberAssignmentKey key.
+     * @param value A ConsumerGroupCurrentMemberAssignmentValue record.
+     */
+    public void replay(
+        ConsumerGroupCurrentMemberAssignmentKey key,
+        ConsumerGroupCurrentMemberAssignmentValue value,
+        GroupType groupType
```

Comment:
All theses group types passes to the replay methods are really confusing too. It gives the impressions that the state is actually persisted but it is not. I think that it will make the code hard to reason about in the future. We should avoid this. Those records are for consumer groups.